File: src/Array/python/haoxiang/FindPeakElement.py
class Solution(object):
    def findPeakElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l = 0
        r = len(nums)-1
        while ((r-l)>1):
            m = (l+r)/2
            if nums[m]>nums[m+1]:
                r = m
            else:
                l = m+1
        return l if (nums[l]>nums[r]) else r

File: src/qiuyi/Array/FindPeakElement.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int m = nums.size();
        int i=0, j=m;

        while (i <= j) {
            int mid = (j-i)/2+i;
            if (mid == 0 && nums[mid] > -1 && nums[mid] > nums[mid+1])
                return mid;
            else if (mid == m-1 && nums[mid] > nums[mid-1])
                return mid;
            if (nums[mid]<nums[mid-1])
                j = mid - 1;
            else if (nums[mid]<nums[mid+1])
                i = mid + 1;
            else
                return mid;
        }
    }
};